benchmark.py

```diff
@@ -0,0 +1,57 @@
+import argparse
+import os
+import subprocess
+
+import utils
+
+def KillBrowsers(browser_list):
+  for browser in browser_list:
+    browser_executable = utils.browsers_definition[browser]['executable']
+    subprocess.call(["killall", "-9", browser_executable])
+
+def Record(scenario_name, driver_script, output_dir, browser=None, extra_args=[], background_script=None):
+
+  with open("powermetrics.plist", "w") as f:
+    powermetrics_process = subprocess.Popen(["sudo", "powermetrics", "-f", "plist", "--samplers", "all", "--show-responsible-pid", "--show-process-gpu", "--show-process-energy", "-i", "60000"], stdout=f, stdin=subprocess.PIPE)
+
+  if browser is not None:
+    browser_executable = utils.browsers_definition[browser]['executable']
+    if browser in ["Chrome", "Canary", "Edge"]:
+      subprocess.call(["open", "-a", browser_executable, "--args"] + extra_args)
+    elif browser == "Safari":
+      subprocess.call(["open", "-a", browser_executable])
+      subprocess.call(["osascript", './driver_scripts/prep_safari.scpt'])
+      subprocess.call(["open", "-a", browser_executable, "--args"] + extra_args)
+
+  if background_script is not None:
+    subprocess.call(["osascript", f'./driver_scripts/{background_script}.scpt'])
+
+  subprocess.call(["osascript", f'./driver_scripts/{driver_script}.scpt'])
+  
+  subprocess.call(["sudo", "killall", "powermetrics"])
```

Comment:
Is it possible to catch ctrl-c and kill powermetrics? It could be done using subprocess instead of killall also.

Reply:
Done for ctrl-c
subprocess.kill doesn't work because sudo is required :(